streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -1775,11 +1735,7 @@ public Map<String, Object> getProducerConfigs(final String clientId) {
         props.putAll(getClientCustomProps());
         props.putAll(clientProvidedProps);
 
-        // When using EOS alpha, stream should auto-downgrade the transactional commit protocol to be compatible with older brokers.
-        if (StreamsConfigUtils.processingMode(this) == StreamsConfigUtils.ProcessingMode.EXACTLY_ONCE_ALPHA) {
-            props.put("internal.auto.downgrade.txn.commit", true);
-        }
-
+        // TODO: given that we remove this code, it seems we can remove the corresponding tests (cf other TODO)
```

Comment:
Open question (we can resolve in this PR). Refers to removing the code above which sets `internal.auto.downgrade.txn.commit`.